File: Sudo_Placement/strings/mystery_number.cpp
#include<bits/stdc++.h>
using namespace std;

int reverseNum(int x)
{
    string s = to_string(x);
    reverse(s.begin(), s.end());
    stringstream ss(s);
    int rev = 0;
    ss>> rev;
    return rev;
}

bool isMystery(int n)
{
    for(int i=1;i<=n/2;i++)
    {
        int j = reverseNum(i);
        if(i+j == n)
        {
            cout<<i<<" "<<j;
            return true;
        }
    }
    cout << "Not a Mystery Number";
    return false;
}

int main()
{
    int n = 121;
    isMystery(n);
    return 0;
}

File: Sudo_Placement/strings/mystery_number.py
def reverseNum(x):
    s = str(x)
    s = s[::-1]
    return int(s);

def isMystery(n):
    for i in range(1, n//2+1):
        j = reverseNum(i)

        if i+j ==n:
            print(i,j)
            return 1
    print("Not a mystery no")
    return 0

n = int(input())
isMystery(n)

